### ISSUE TITLE
Could not create proposal in mercator 

### ISSUE THREAD

**Issue by MagdaN** [liqd/adhocracy3.mercator#1135]
Error: "InterfaceClass adhocracy_mercator.resources.mercator.IMercatorProposal is not one of InterfaceClass adhocracy_core.resources.external_resource.IExternalResource

As I could not create any proposals I was not able to test mercator.

**Comment by xi** [comment]
See discussion in #1119

> For Mercator the permissions are set in resource/subscriber.py and reflect the current status of the platform i.e it's not possible to create proposal anymore now.

**Comment by xi** [comment]
`resource/subscriber.py` defines that god has permission to do anything. So @pallix' answer can not be the full truth.

Error message:

```
"<InterfaceClass adhocracy_mercator.resources.mercator.IMercatorProposal>" is not one of <InterfaceClass adhocracy_core.resources.pool.IBasicPool>, <InterfaceClass adhocracy_core.resources.external_resource.IExternalResource>, <InterfaceClass adhocracy_core.interfaces.IPool>, <InterfaceClass adhocracy_core.resources.asset.IPoolWithAssets>
```

**Comment by pallix** [comment]
there is still the evolution script ``` evolve4_disable_voting_and_commenting```  in the code base. With which user are you creating the proposal?

**Comment by xi** [comment]
> there is still the evolution script evolve4_disable_voting_and_commenting in the code base.

I actually ran the migrations. But that is also necessary in order to use the old database with the current code base.

> With which user are you creating the proposal?

god

**Comment by pallix** [comment]
Ok i can reproduce it with god. I get this error in the log

```
2015-05-13 18:23:19,978 WARNI [adhocracy_core.rest.views][MainThread] Found 1 validation errors in request: <b'{"parent": "http://localhost:6541/mercator", "content_type": "adhocracy_mercator.resources.mercator.IMercatorProposal", "first_version_path": "@pn52", "data": {"adhocracy_core.sheets.name.IName": {"name": "x701"}}, "root_versions": []}'>
2015-05-13 18:23:19,978 WARNI [adhocracy_core.rest.views][MainThread]   {'location': 'body', 'description': '"<InterfaceClass adhocracy_mercator.resources.mercator.IMercatorProposal>" is not one of <InterfaceClass adhocracy_core.interfaces.IPool>, <InterfaceClass adhocracy_core.resources.external_resource.IExternalResource>, <InterfaceClass adhocracy_core.resources.pool.IBasicPool>, <InterfaceClass adhocracy_core.resources.asset.IPoolWithAssets>', 'name': 'content_type'}
```

**Comment by pallix** [comment]
Somehow the `request.has_permission(permission, context)` is false for `add_proposal` even if `('Allow', 'role:god', 'add_proposal')` is defined in the root ACL.

**Comment by pallix** [comment]
@joka 

During the request the following principals are active:

```
{'role:annotator', 'role:contributor', 'group:gods', 'role:god', 'system.Everyone',                       
'/principals/users/0000000', 'system.Authenticated', 'role:reader', 'group:authenticated'} 
```

Since god does not have anymore the ALL_PERMISSIONS in front of the acl list (since we introduce ACM) and 'role:contribuator' has a deny for 'add_proposal' the request fails.

One solution would be to remove the definition of the permissions for god in the  various ACM and always ensures ALL_PERMISSIONS is in front of the acl list.

Also we need some functional tests even for the god user to avoid breaking the client again.

**Comment by pallix** [comment]
@xi, @MagdaN  As a temporary measure you can set the permission to Allow in ```resource/mercator.py``` in the ```mercator_acm``` but please do not commit your change.

**Comment by pallix** [comment]
Fixes applied in #1146

**Comment by pallix** [comment]
Merged
